app/controllers/application_controller.rb

```diff
@@ -1,36 +1,31 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::API
-    include ActionController::HttpAuthentication::Token::ControllerMethods
-   
-    # Add a before_action to authenticate all requests.
-    # Move this to subclassed controllers if you only
-    # want to authenticate certain methods.
-    #rescue_from ActiveRecord::RecordNotFound, :with => :render_404
-    #rescue_from AbstractController::ActionNotFound, :with => :render_404
+  include ActionController::HttpAuthentication::Token::ControllerMethods
 
-    before_action :authenticate
-  
-    protected
-  
-    # Authenticate the user with token based authentication
-    def authenticate
-      authenticate_token || render_unauthorized
-    end
-  
-    def authenticate_token
-      authenticate_with_http_token do |token_key, options|
-        token_key = token_key.split(/(?:api_key=)(\w*)(?: api_secret=)(\w*)/)
-        @current_bot = Bot.select(:id, :username, :name, :bio, :created_at).find_by(api_key: token_key[1], api_secret: token_key[2])
-      end
-    end
-  
-    def render_unauthorized(realm = "Application")
-      self.headers["WWW-Authenticate"] = %(Token_id realm="#{realm}")
-      render json: {status: 'ERROR', message:'401 - bad credentials'}, status: :unauthorized
-    end
+  before_action :authenticate
+
+  protected
 
-    def render_404
-      render json: {status: 'ERROR', message:'404 - record not found'}, status: :not_found
+  # Authenticate the user with token based authentication
+  def authenticate
+    authenticate_token || render_unauthorized
+  end
+
+  def authenticate_token
+    authenticate_with_http_token do |token_key, _options|
+      token_key = token_key.split(/api_key=(\w*) api_secret=(\w*)/)
```

Comment:
Isn't it a good idea to change this ragex? We already laughed about it, we just need to think of a better one ;)

PS: A issue to discuss is: is this safe? Perhaps the best way is to get the bot to authenticate with the Device. That's how it is on twitter (with Auth0). Did we do it this way because it is faster or easier?